### PR TITLE
chore(release): fix release automation issues

### PR DESCRIPTION
**Pull request by bakoushin**
### Description

Follow-up to #5879. Various fixes to make it work.

### Test plan

Tested on a branch: https://github.com/valora-inc/wallet/actions/runs/10702029039 with successful iOS nightly build.
The Android build failure is likely due to Google API not respecting the API 33 deadline extension, to be investigated.
Android build is currently failing in the release automation repo as well.

### Related issues

- Related to RET-1161

### Backwards compatibility

NA

### Network scalability

NA

### CODE REVIEW

File: .github/workflows/release-nightly.yml [hunk 1]
```diff
@@ -20,19 +20,19 @@ jobs:
         uses: actions/checkout@v4
       - id: latest-commit-hash
         name: Print latest commit
-        run: echo $(git rev-parse HEAD) && echo "::set-output name=latest-commit-hash::$(git rev-parse HEAD)"
+        run: echo $(git rev-parse HEAD) && echo "latest-commit-hash=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
```
Comment (bakoushin) [inline .github/workflows/release-nightly.yml:23]: this syntax was deprecated and triggered a warning.
replaced with a [recommended approach](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).

File: .github/workflows/release-fastlane-android.yml [hunk 2]
```diff
@@ -24,8 +24,12 @@ jobs:
...
         lane: ${{ fromJson(inputs.lanes) }}
     steps:
-      - uses: actions/checkout@v4
-      - uses: google-github-actions/auth@v2
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
```
Comment (bakoushin) [inline .github/workflows/release-fastlane-android.yml:32]: checkout must go first so google auth token is available in the subsequent steps.
see docs: https://github.com/google-github-actions/auth#prerequisites

the next checkouts will erase the auth token, so I reduced the steps to a single checkout.

File: Gemfile.lock [hunk 3]
```diff
@@ -156,9 +156,9 @@ GEM
       xcodeproj (>= 1.13.0, < 2.0.0)
       xcpretty (~> 0.3.0)
       xcpretty-travis-formatter (>= 0.0.3)
-    fastlane-plugin-emerge (0.5.0)
+    fastlane-plugin-emerge (0.10.4)
```
Comment (bakoushin) [inline Gemfile.lock:159]: updating fastlane libraries as they recommend in the logs:
```
+--------------------------------------------+
|          Plugin updates available          |
+------------+--------------+----------------+
| Plugin     | Your Version | Latest Version |
+------------+--------------+----------------+
| versioning | 0.4.3        | 0.6.0          |
| emerge     | 0.5.0        | 0.10.4         |
+------------+--------------+----------------+
[13:08:45]: To update all plugins, just run
[13:08:45]: $ bundle exec fastlane update_plugins
```

File: .github/workflows/release-fastlane-android.yml [hunk 4]
```diff
@@ -44,15 +48,10 @@ jobs:
       - uses: webfactory/ssh-agent@v0.9.0
         with:
           ssh-private-key: ${{ steps.google-secrets.outputs.BOT_SSH_KEY }}
```
Comment (jeanregisser) [inline .github/workflows/release-fastlane-android.yml:50]: Cool, I think we can remove this one too. It was needed when we had the branding.

**Review by jeanregisser — APPROVED** [review]
Thanks! 🚀

**Pull request merged**